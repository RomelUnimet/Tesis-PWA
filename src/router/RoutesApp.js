import React, { useState} from 'react'
import { Switch, Route, Redirect } from 'react-router-dom';


import { CardScreen } from '../components/cards/CardScreen';
import { CreateModal } from '../components/create_entry/CreateModal';
import { ProfileScreen } from '../components/profile/ProfileScreen';
import { CardEntries } from '../components/cards/CardEntries';
import { Navbar } from '../components/ui/Navbar'
import { SettingsScreen } from '../components/settings/SettingsScreen';



export const RoutesApp = () => {

    //CAMBIE LOS STORES DE ESTA RUTA A LA RUTA PRINCIPAL PARA QUE NO SE LLAMARAN TANTO
    //SI HAY UN ERROR CON ESO ES MUY PROBABLE QUE SEA AHI

    const [CEModalState, setCEModalState] = useState(false);

    //ESTO ES LO QUE ME ESTA CAUSANDO PROBLEMAS
    //CON CADA CAMBIO DE RUTA SE VUELVE A INICIALIZAR
    const [usedNavbar, setusedNavbar] = useState('') 

    return (
        <>
            
            <CreateModal
                CEModalState={CEModalState}
                setCEModalState={setCEModalState}
            />
        
            <div 
                style={{width:'100vw', height:'100vh', overflow:'hidden', position:'absolute'}}
            >
                
                    <Switch >
                        <Route
                            exact
                            path="/cards" 
                        >
                            <CardScreen/>
                        </Route>
                           
                        <Route
                            exact
                            path="/detailedcard/:id" 
                        >
                            <CardEntries
                                usedNavbar={usedNavbar}
                            />
                        </Route>
                            
                        <Route
                            exact
                            path="/profile" 
                        >
                            <ProfileScreen/>
                        </Route>
                        <Route
                            exact
                            path="/settings" 
                        >
                            <SettingsScreen
                                usedNavbar={usedNavbar}
                            />
                        </Route>
                            
                        <Redirect to="/cards"/>

                    </Switch>
            </div>    

            <Navbar
                CEModalState={CEModalState}
                setCEModalState={setCEModalState}
                setusedNavbar={setusedNavbar}
            />
            
            
            
        </>
    )

    
}
