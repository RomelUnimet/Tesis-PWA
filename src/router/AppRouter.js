import React, {useEffect} from 'react';
import {
    Switch,
    Redirect,
    useLocation
  } from 'react-router-dom';
  
import { useDispatch, useSelector } from 'react-redux';

import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';

import { RegisterScreen } from '../components/auth/RegisterScreen'
import { startChecking } from '../actions/auth';
import { RoutesApp } from './RoutesApp';

import { AnimatePresence } from "framer-motion";

import { startCardStore } from '../actions/cards';
import { startSettingsStore } from '../actions/settings';
import { startGetWeather } from '../actions/extra';
import { startTagStore } from '../actions/tag'
import { startLocationStore } from '../actions/location'
import { startEntryStore } from '../actions/entry'

import Localbase from 'localbase';
import { creteOptionsObject } from '../helpers/createOptionsCredential';

import { askForPermissionToReceiveNotifications } from '../notification';


const db = new Localbase('pwa-card-diary');


export const AppRouter = () => {

    const dispatch = useDispatch();

    useEffect(  () => {
    
        dispatch( startChecking() );
        dispatch( startSettingsStore() );
        dispatch( startCardStore() );
        dispatch( startTagStore() );
        dispatch( startLocationStore() );
        dispatch( startEntryStore() );
        dispatch( startGetWeather() );

        //PROBAR QUE ESTO NO HACE QUE DEJE DE FUNCIONAR
        //NECESITO TAMBIEN COLOCAR ESTO AQUI PARA HACER EL CHECK CADA VEZ QUE SE ABRE LA APP
        askForPermissionToReceiveNotifications()
        
    }, [dispatch])

    useEffect(() => {
        //Funcion para trigger la Autenticacion de la pagina
        async function runAuth() {
            
            const [userSettings] = await db.collection('userSettings').get();

            if( !!userSettings && userSettings.auth && navigator.credentials && window.PublicKeyCredential){ //si no esta inicuado fa ettor
                //PONER A TRIGGER AUTH COMO UN HELPER QUE SE USE EN LOCK COMO AQUI
                try {

                    const optionsFromServer = creteOptionsObject()
        
                    const credential = await navigator.credentials.create({
                        publicKey: optionsFromServer 
                    });
        
                    console.log(credential)
        
                } catch (error) {
        
                    alert('WebAuthn Doesnt Allow to call Create if its not triggered by an user action')
                    alert(error)
                    
                }
            }
        }
                
        runAuth();
    }, [])



    const { checking, uid } = useSelector( state => state.auth);
  
    const location = useLocation();


    if ( checking ) {
        return (<h5>Espere...</h5>);
        //Componente de espera
    }

    return (
        
            <div>
                <AnimatePresence
                    
                    initial={false}
                >

                    <Switch location={location} key={location.pathname}>

                        <PublicRoute 
                            exact 
                            path="/auth" 
                            isAuthenticated={ !!uid }
                            component={RegisterScreen}
                        />

                        <PrivateRoute 
                            exact 
                            path="" 
                            isAuthenticated={ !!uid }
                            component={RoutesApp}
                        />
                        <Redirect to="" />   
                    </Switch>
                </AnimatePresence>

            </div>
        
    )
}