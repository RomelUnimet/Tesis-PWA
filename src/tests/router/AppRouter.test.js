import { mount, configure } from "enzyme"
import { Provider } from 'react-redux'
import { store } from '../../store/store'
import { AppRouter } from "../../router/AppRouter";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
      pathname: "localhost:3000/"
    })
}));
  


configure({ adapter: new Adapter() })

const middlewares = [thunk]

const mockStore = configureStore(middlewares)

const initState = {
  auth:{
    checking:false,
    uid:"6206bbb958922200161ea0c3"
  }
}

let reduxStore = mockStore(initState)

reduxStore.dispatch = jest.fn()

describe('Pruebas app router', () => { 

    
    
    test('should render the router', async () => { 

        
        const wrapper = mount(
            <Provider store={store}>
                <AppRouter/>
            </Provider>
        )

        console.log(wrapper.html())
        expect(wrapper).toMatchSnapshot()

        
     })

 })