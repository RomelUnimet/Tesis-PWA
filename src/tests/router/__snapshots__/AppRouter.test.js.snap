// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas app router should render the router 1`] = `ReactWrapper {}`;
